lib/services/product-manager.js

```diff
@@ -55,6 +55,21 @@ export default class ProductManager {
     return this._filterOutDuplicateProducts(_.flatten(productBatches))
   }
 
+  async getProductsBySkusOrSlugs (skus, slugs) {
+    // filter out duplicates and divide skus into chunks
+    const skuChunks = _(skus).uniq().chunk(this.loadBatchCount).value()
+    const slugChunks = _(slugs).uniq().chunk(this.loadBatchCount).value()
+
+    const productBatches = await Promise.map(skuChunks,
+      (skuChunk, i) => {
+        const slugChunk = slugChunks[i]
```

Comment:
I found out that you check if the slugs chunk exist before you append them to the predicate using condition `slugs.length` but if you take eg. `slugChunks[8298398]` it will return undefined so you will possibly check the length of `undefined` value.

Also, I still think that if you have more SKUs than slugs it is faster, for example, to read in parallel 50 SKUs per chunk and separately for example 3 slugs per chunk then combine it together and load 2 products per chunk

When you have 100 skus in 24 products (= 24 slugs):
 - loading it separately means: 2 request for loading SKUs and in parallel 24 / 3  = 8 requests for loading slugs so the loading time is `max(2 SKU requests, 8 slug requests)` and there can be used some concurrency to speed it up
 - loading it together means smaller chunks - eg split by 2: 100 / 2 = 50 requests and again some concurrency

Reply:
> I found out that you check if the slugs chunk exist before you append them to the predicate using condition slugs.length but if you take eg. slugChunks[8298398] it will return undefined so you will possibly check the length of undefined value.
Do you mean in the method `_getProductsBySkuOrSlugChunk (skus, slugs = [])`? If `slugChunk` is undefined, it will get the default value `[]` from the method param.

Reply:
https://github.com/commercetools/commercetools-node-variant-reassignment/pull/45/commits/c1205ac140a9494165831e3c493b495623b38ac4

Reply:
After discussion with Roman we had a conclusion that this approach is also not optimal, but for now, I created a task for future: https://github.com/commercetools/commercetools-node-variant-reassignment/issues/53